src/math/ec_keys.cpp

```diff
@@ -128,7 +162,7 @@ ec_secret create_nonce(ec_secret secret, hash_digest hash)
     {
         V = hmac_sha256_hash(V, K);
 
-        if (verify_private_key(V))
+        if (0 == index--)
```

Comment:
It's the mixing of mutation and comparison that I feel is confusing, not
necessarily the use of the postfix op in general.

On 6 Jan 2015 10:03 am, "William Swanson" <notifications@github.com> wrote:
>
> In src/math/ec_keys.cpp:
>
> > @@ -128,7 +162,7 @@ ec_secret create_nonce(ec_secret secret,
hash_digest hash)
> >      {
> >          V = hmac_sha256_hash(V, K);
> >
> > -        if (verify_private_key(V))
> > +        if (0 == index--)
>
> Maybe, but unless you are a Python developer, the ++ and -- operators are
a perfectly common, everyday operators. I think we can assume that people
reading this code are fairly familiar with the language. Otherwise, I
suppose we could do this:
>
> if (0 == index)
>     return V;
> --index;
>
> —
> Reply to this email directly or view it on GitHub.